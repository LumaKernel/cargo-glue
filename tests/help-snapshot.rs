use insta::assert_snapshot;
use std::{process::Output, str};

#[test]
fn helps() -> anyhow::Result<()> {
    assert_snapshot!("short", run("-h")?);
    assert_snapshot!("long", run("--help")?);
    Ok(())
}

fn run(flag: &str) -> anyhow::Result<String> {
    let assert = assert_cmd::Command::cargo_bin("cargo-glue")?
        .args(["glue", flag])
        .assert()
        .success();
    let Output { stdout, .. } = assert.get_output();
    Ok(str::from_utf8(stdout)?.replacen(env!("CARGO_PKG_VERSION"), "<version>", 1))
}
