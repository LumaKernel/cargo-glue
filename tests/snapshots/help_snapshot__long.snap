---
source: tests/help-snapshot.rs
expression: "run(\"--help\")?"
---
cargo-glue <version>
Ryo Yamashita <qryxip@gmail.com>

A Cargo subcommand to bundle your code into one `.rs` file for competitive programming.

Use -h for short descriptions and --help for more detials.

USAGE:
    cargo glue [OPTIONS]
    cargo glue [OPTIONS] --lib
    cargo glue [OPTIONS] --bin <NAME>
    cargo glue [OPTIONS] --example <NAME>
    cargo glue [OPTIONS] --src <PATH>

OPTIONS:
        --src <PATH>                                  
            Bundle the lib/bin/example target and its dependencies.
            
            This option is intended to be used from editors such as VSCode. Use `--lib`, `--bin` or `--example` for
            normal usage.
        --lib                                         
            Bundle the library and its dependencies

        --bin <NAME>                                  
            Bundle the binary and its dependencies

        --example <NAME>                              
            Bundle the binary example and its dependencies

        --manifest-path <PATH>                        
            Path to Cargo.toml

        --exclude <SPEC>...                           
            Exclude library crates from bundling

        --exclude-atcoder-crates                      
            Alias for:
            --exclude https://github.com/rust-lang/crates.io-index#alga:0.9.3
                      https://github.com/rust-lang/crates.io-index#ascii:1.0.0
                      https://github.com/rust-lang/crates.io-index#bitset-fixed:0.1.0
                      https://github.com/rust-lang/crates.io-index#either:1.5.3
                      https://github.com/rust-lang/crates.io-index#fixedbitset:0.2.0
                      https://github.com/rust-lang/crates.io-index#getrandom:0.1.14
                      https://github.com/rust-lang/crates.io-index#im-rc:14.3.0
                      https://github.com/rust-lang/crates.io-index#indexmap:1.3.2
                      https://github.com/rust-lang/crates.io-index#itertools:0.9.0
                      https://github.com/rust-lang/crates.io-index#itertools-num:0.1.3
                      https://github.com/rust-lang/crates.io-index#lazy_static:1.4.0
                      https://github.com/rust-lang/crates.io-index#libm:0.2.1
                      https://github.com/rust-lang/crates.io-index#maplit:1.0.2
                      https://github.com/rust-lang/crates.io-index#nalgebra:0.20.0
                      https://github.com/rust-lang/crates.io-index#ndarray:0.13.0
                      https://github.com/rust-lang/crates.io-index#num:0.2.1
                      https://github.com/rust-lang/crates.io-index#num-bigint:0.2.6
                      https://github.com/rust-lang/crates.io-index#num-complex:0.2.4
                      https://github.com/rust-lang/crates.io-index#num-derive:0.3.0
                      https://github.com/rust-lang/crates.io-index#num-integer:0.1.42
                      https://github.com/rust-lang/crates.io-index#num-iter:0.1.40
                      https://github.com/rust-lang/crates.io-index#num-rational:0.2.4
                      https://github.com/rust-lang/crates.io-index#num-traits:0.2.11
                      https://github.com/rust-lang/crates.io-index#ordered-float:1.0.2
                      https://github.com/rust-lang/crates.io-index#permutohedron:0.2.4
                      https://github.com/rust-lang/crates.io-index#petgraph:0.5.0
                      https://github.com/rust-lang/crates.io-index#proconio:0.3.6
                      https://github.com/rust-lang/crates.io-index#proconio:0.3.7
                      https://github.com/rust-lang/crates.io-index#proconio:0.3.8
                      https://github.com/rust-lang/crates.io-index#rand:0.7.3
                      https://github.com/rust-lang/crates.io-index#rand_chacha:0.2.2
                      https://github.com/rust-lang/crates.io-index#rand_core:0.5.1
                      https://github.com/rust-lang/crates.io-index#rand_distr:0.2.2
                      https://github.com/rust-lang/crates.io-index#rand_hc:0.2.0
                      https://github.com/rust-lang/crates.io-index#rand_pcg:0.2.1
                      https://github.com/rust-lang/crates.io-index#regex:1.3.6
                      https://github.com/rust-lang/crates.io-index#rustc-hash:1.1.0
                      https://github.com/rust-lang/crates.io-index#smallvec:1.2.0
                      https://github.com/rust-lang/crates.io-index#superslice:1.0.0
                      https://github.com/rust-lang/crates.io-index#text_io:0.1.8
                      https://github.com/rust-lang/crates.io-index#whiteread:0.5.0
             
        --exclude-atcoder-202301-crates               
            Alias for:
            --exclude https://github.com/rust-lang/crates.io-index#ac-library-rs:0.1.1
                      https://github.com/rust-lang/crates.io-index#once_cell:1.18.0
                      https://github.com/rust-lang/crates.io-index#static_assertions:1.1.0
                      https://github.com/rust-lang/crates.io-index#varisat:0.2.2
                      https://github.com/rust-lang/crates.io-index#memoise:0.3.2
                      https://github.com/rust-lang/crates.io-index#argio:0.2.0
                      https://github.com/rust-lang/crates.io-index#bitvec:1.0.1
                      https://github.com/rust-lang/crates.io-index#counter:0.5.7
                      https://github.com/rust-lang/crates.io-index#hashbag:0.1.11
                      https://github.com/rust-lang/crates.io-index#pathfinding:4.3.0
                      https://github.com/rust-lang/crates.io-index#recur-fn:2.2.0
                      https://github.com/rust-lang/crates.io-index#indexing:0.4.1
                      https://github.com/rust-lang/crates.io-index#amplify:3.14.2
                      https://github.com/rust-lang/crates.io-index#amplify_derive:2.11.3
                      https://github.com/rust-lang/crates.io-index#amplify_num:0.4.1
                      https://github.com/rust-lang/crates.io-index#easy-ext:1.0.1
                      https://github.com/rust-lang/crates.io-index#multimap:0.9.0
                      https://github.com/rust-lang/crates.io-index#btreemultimap:0.1.1
                      https://github.com/rust-lang/crates.io-index#bstr:1.6.0
                      https://github.com/rust-lang/crates.io-index#az:1.2.1
                      https://github.com/rust-lang/crates.io-index#glidesort:0.1.2
                      https://github.com/rust-lang/crates.io-index#tap:1.0.1
                      https://github.com/rust-lang/crates.io-index#omniswap:0.1.0
                      https://github.com/rust-lang/crates.io-index#multiversion:0.7.2
                      https://github.com/rust-lang/crates.io-index#num:0.4.1
                      https://github.com/rust-lang/crates.io-index#num-bigint:0.4.3
                      https://github.com/rust-lang/crates.io-index#num-complex:0.4.3
                      https://github.com/rust-lang/crates.io-index#num-integer:0.1.45
                      https://github.com/rust-lang/crates.io-index#num-iter:0.1.43
                      https://github.com/rust-lang/crates.io-index#num-rational:0.4.1
                      https://github.com/rust-lang/crates.io-index#num-traits:0.2.15
                      https://github.com/rust-lang/crates.io-index#num-derive:0.4.0
                      https://github.com/rust-lang/crates.io-index#ndarray:0.15.6
                      https://github.com/rust-lang/crates.io-index#nalgebra:0.32.3
                      https://github.com/rust-lang/crates.io-index#alga:0.9.3
                      https://github.com/rust-lang/crates.io-index#libm:0.2.7
                      https://github.com/rust-lang/crates.io-index#rand:0.8.5
                      https://github.com/rust-lang/crates.io-index#getrandom:0.2.10
                      https://github.com/rust-lang/crates.io-index#rand_chacha:0.3.1
                      https://github.com/rust-lang/crates.io-index#rand_core:0.6.4
                      https://github.com/rust-lang/crates.io-index#rand_hc:0.3.2
                      https://github.com/rust-lang/crates.io-index#rand_pcg:0.3.1
                      https://github.com/rust-lang/crates.io-index#rand_distr:0.4.3
                      https://github.com/rust-lang/crates.io-index#petgraph:0.6.3
                      https://github.com/rust-lang/crates.io-index#indexmap:2.0.0
                      https://github.com/rust-lang/crates.io-index#regex:1.9.1
                      https://github.com/rust-lang/crates.io-index#lazy_static:1.4.0
                      https://github.com/rust-lang/crates.io-index#ordered-float:3.7.0
                      https://github.com/rust-lang/crates.io-index#ascii:1.1.0
                      https://github.com/rust-lang/crates.io-index#permutohedron:0.2.4
                      https://github.com/rust-lang/crates.io-index#superslice:1.0.0
                      https://github.com/rust-lang/crates.io-index#itertools:0.11.0
                      https://github.com/rust-lang/crates.io-index#itertools-num:0.1.3
                      https://github.com/rust-lang/crates.io-index#maplit:1.0.2
                      https://github.com/rust-lang/crates.io-index#either:1.8.1
                      https://github.com/rust-lang/crates.io-index#im-rc:15.1.0
                      https://github.com/rust-lang/crates.io-index#fixedbitset:0.4.2
                      https://github.com/rust-lang/crates.io-index#bitset-fixed:0.1.0
                      https://github.com/rust-lang/crates.io-index#proconio:0.4.5
                      https://github.com/rust-lang/crates.io-index#text_io:0.1.12
                      https://github.com/rust-lang/crates.io-index#rustc-hash:1.1.0
                      https://github.com/rust-lang/crates.io-index#smallvec:1.11.0
             
        --exclude-atcoder-202308-crates               
            Alias for:
            --exclude https://github.com/rust-lang/crates.io-index#ac-library-rs:0.1.1
                      https://github.com/rust-lang/crates.io-index#once_cell:1.18.0
                      https://github.com/rust-lang/crates.io-index#static_assertions:1.1.0
                      https://github.com/rust-lang/crates.io-index#varisat:0.2.2
                      https://github.com/rust-lang/crates.io-index#memoise:0.3.2
                      https://github.com/rust-lang/crates.io-index#argio:0.2.0
                      https://github.com/rust-lang/crates.io-index#bitvec:1.0.1
                      https://github.com/rust-lang/crates.io-index#counter:0.5.7
                      https://github.com/rust-lang/crates.io-index#hashbag:0.1.11
                      https://github.com/rust-lang/crates.io-index#pathfinding:4.3.0
                      https://github.com/rust-lang/crates.io-index#recur-fn:2.2.0
                      https://github.com/rust-lang/crates.io-index#indexing:0.4.1
                      https://github.com/rust-lang/crates.io-index#amplify:3.14.2
                      https://github.com/rust-lang/crates.io-index#amplify_derive:2.11.3
                      https://github.com/rust-lang/crates.io-index#amplify_num:0.4.1
                      https://github.com/rust-lang/crates.io-index#easy-ext:1.0.1
                      https://github.com/rust-lang/crates.io-index#multimap:0.9.0
                      https://github.com/rust-lang/crates.io-index#btreemultimap:0.1.1
                      https://github.com/rust-lang/crates.io-index#bstr:1.6.0
                      https://github.com/rust-lang/crates.io-index#az:1.2.1
                      https://github.com/rust-lang/crates.io-index#glidesort:0.1.2
                      https://github.com/rust-lang/crates.io-index#tap:1.0.1
                      https://github.com/rust-lang/crates.io-index#omniswap:0.1.0
                      https://github.com/rust-lang/crates.io-index#multiversion:0.7.2
                      https://github.com/rust-lang/crates.io-index#num:0.4.1
                      https://github.com/rust-lang/crates.io-index#num-bigint:0.4.3
                      https://github.com/rust-lang/crates.io-index#num-complex:0.4.3
                      https://github.com/rust-lang/crates.io-index#num-integer:0.1.45
                      https://github.com/rust-lang/crates.io-index#num-iter:0.1.43
                      https://github.com/rust-lang/crates.io-index#num-rational:0.4.1
                      https://github.com/rust-lang/crates.io-index#num-traits:0.2.15
                      https://github.com/rust-lang/crates.io-index#num-derive:0.4.0
                      https://github.com/rust-lang/crates.io-index#ndarray:0.15.6
                      https://github.com/rust-lang/crates.io-index#nalgebra:0.32.3
                      https://github.com/rust-lang/crates.io-index#alga:0.9.3
                      https://github.com/rust-lang/crates.io-index#libm:0.2.7
                      https://github.com/rust-lang/crates.io-index#rand:0.8.5
                      https://github.com/rust-lang/crates.io-index#getrandom:0.2.10
                      https://github.com/rust-lang/crates.io-index#rand_chacha:0.3.1
                      https://github.com/rust-lang/crates.io-index#rand_core:0.6.4
                      https://github.com/rust-lang/crates.io-index#rand_hc:0.3.2
                      https://github.com/rust-lang/crates.io-index#rand_pcg:0.3.1
                      https://github.com/rust-lang/crates.io-index#rand_distr:0.4.3
                      https://github.com/rust-lang/crates.io-index#petgraph:0.6.3
                      https://github.com/rust-lang/crates.io-index#indexmap:2.0.0
                      https://github.com/rust-lang/crates.io-index#regex:1.9.1
                      https://github.com/rust-lang/crates.io-index#lazy_static:1.4.0
                      https://github.com/rust-lang/crates.io-index#ordered-float:3.7.0
                      https://github.com/rust-lang/crates.io-index#ascii:1.1.0
                      https://github.com/rust-lang/crates.io-index#permutohedron:0.2.4
                      https://github.com/rust-lang/crates.io-index#superslice:1.0.0
                      https://github.com/rust-lang/crates.io-index#itertools:0.11.0
                      https://github.com/rust-lang/crates.io-index#itertools-num:0.1.3
                      https://github.com/rust-lang/crates.io-index#maplit:1.0.2
                      https://github.com/rust-lang/crates.io-index#either:1.8.1
                      https://github.com/rust-lang/crates.io-index#im-rc:15.1.0
                      https://github.com/rust-lang/crates.io-index#fixedbitset:0.4.2
                      https://github.com/rust-lang/crates.io-index#bitset-fixed:0.1.0
                      https://github.com/rust-lang/crates.io-index#proconio:0.4.5
                      https://github.com/rust-lang/crates.io-index#text_io:0.1.12
                      https://github.com/rust-lang/crates.io-index#rustc-hash:1.1.0
                      https://github.com/rust-lang/crates.io-index#smallvec:1.11.0
             
        --exclude-codingame-crates                    
            Alias for:
            --exclude https://github.com/rust-lang/crates.io-index#chrono:0.4.19
                      https://github.com/rust-lang/crates.io-index#itertools:0.10.0
                      https://github.com/rust-lang/crates.io-index#libc:0.2.93
                      https://github.com/rust-lang/crates.io-index#rand:0.8.3
                      https://github.com/rust-lang/crates.io-index#regex:1.4.5
                      https://github.com/rust-lang/crates.io-index#time:0.2.26
             
        --mine <DOMAIN_AND_USERNAME>...               
            Do not include license and copyright notices for the users.
            
            Supported formats:
            * github.com/{username}
            * gitlab.com/{username}
             
        --toolchain-for-udeps <TOOLCHAIN>             
            `nightly` toolchain for `cargo-udeps` [default: nightly]

        --toolchain-for-proc-macro-srv <TOOLCHAIN>    
            Toolchain for expanding procedural macros

        --mod-path <MODULE_PATH>                      
            Expand the libraries to the module [default: crate::__cargo_glue]

        --remove <REMOVE>...                          
            Removes
            * doc comments (`//! ..`, `/// ..`, `/** .. */`, `#[doc = ".."]`) with `--remove docs`.
            * comments (`// ..`, `/* .. */`) with `--remove comments`.
            
            ```
            #[allow(dead_code)]
            pub mod a {
                //! A.
            
                /// A.
                pub struct A; // aaaaa
            }
            ```
            
            ↓
            
            ```
            #[allow(dead_code)]
            pub mod a {
                pub struct A;
            }
            ```
             
        --minify <MINIFY>
            Minifies
            - each expaned library with `--minify lib`.
            - the whole code with `--minify all`.
            
            Not that the minification function is incomplete. Unnecessary spaces may be inserted.
             
        --no-resolve-cfgs                             
            Do not resolve `cfg(..)`s

        --no-rustfmt                                  
            Do not format the output before emitting

        --no-check                                    
            Do not check the output before emitting

    -o, --output <PATH>                               
            Write to the file instead of STDOUT

    -h, --help                                        
            Prints help information

    -V, --version                                     
            Prints version information
